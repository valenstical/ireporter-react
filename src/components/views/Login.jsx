import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import loadingIcon from '../../../public/images/resources/loader-light.gif';
import Alert from './Alert';

/**
 * The presentational component for the login page
 * @param {object} props An object containing the required properties of the component
 * @returns {object} The generated JSX object
 */
export default function Login({ user, handleSubmit }) {
  return (
    <section className="how login-wrapper bg-grey top-space" data-pg-name="How it Works">
      <div className="wrapper" data-pg-name="Wrapper">
        <form className="card login-form" noValidate onSubmit={handleSubmit}>
          <header className="form-header">
            <h1 className="text-brand">Welcome Back</h1>
            <p>Sign in to start reporting...</p>
          </header>
          {user.message.length > 0 && <Alert message={user.message} title={user.success ? 'Login Successful!' : 'Login Failed!'} success={user.success} />}
          <div className="form-wrapper">
            <label htmlFor="username">Username, Email or Phone number</label>
            <input className="form-element" type="text" name="username" required id="username" />
          </div>
          <div className="form-wrapper">
            <label htmlFor="forgotPassword">Password</label>
            <Link to="#" className="forgot-password transition">Forgot Password?</Link>
            <input className="form-element" type="password" name="password" required id="forgotPassword" autoComplete="off" />
          </div>
          <button className="btn-brand btn-submit" type="submit" disabled={user.isBusy}>
            <span>Sign In </span>
            { user.isBusy && <img src={loadingIcon} width="12" alt="loading" />}
          </button>
        </form>
        <p className="login-register-link">
New to iReporter?
          {' '}
          <Link to="/sign-up">Register here</Link>
        </p>
      </div>
    </section>


  );
}

Login.propTypes = {
  user: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};
