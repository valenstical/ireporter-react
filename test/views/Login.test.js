import React from 'react';
import { shallow } from 'enzyme';
import Login from '../../src/components/views/Login';

const user = {
  isBusy: false,
  success: false,
  message: [],
};

describe('<Login>', () => {
  it('should render without crashing on startup', () => {
    const wrapper = shallow(<Login
      user={user}
      handleSubmit={jest.fn()}
      />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing when login failed', () => {
    const wrapper = shallow(<Login user={{ isBusy: false, success: false, message: ['failed'] }} handleSubmit={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing when login is successful', () => {
    const wrapper = shallow(<Login user={{ isBusy: true, success: true, message: ['success'] }} handleSubmit={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
