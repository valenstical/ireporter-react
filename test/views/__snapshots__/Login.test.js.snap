// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login> should render without crashing on startup 1`] = `
<section
  className="how login-wrapper bg-grey top-space"
  data-pg-name="How it Works"
>
  <div
    className="wrapper"
    data-pg-name="Wrapper"
  >
    <form
      className="card login-form"
      noValidate={true}
      onSubmit={[MockFunction]}
    >
      <header
        className="form-header"
      >
        <h1
          className="text-brand"
        >
          Welcome Back
        </h1>
        <p>
          Sign in to start reporting...
        </p>
      </header>
      <div
        className="form-wrapper"
      >
        <label
          htmlFor="username"
        >
          Username, Email or Phone number
        </label>
        <input
          className="form-element"
          id="username"
          name="username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-wrapper"
      >
        <label
          htmlFor="forgotPassword"
        >
          Password
        </label>
        <Link
          className="forgot-password transition"
          replace={false}
          to="#"
        >
          Forgot Password?
        </Link>
        <input
          autoComplete="off"
          className="form-element"
          id="forgotPassword"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn-brand btn-submit"
        disabled={false}
        type="submit"
      >
        <span>
          Sign In 
        </span>
      </button>
    </form>
    <p
      className="login-register-link"
    >
      New to iReporter?
       
      <Link
        replace={false}
        to="/sign-up"
      >
        Register here
      </Link>
    </p>
  </div>
</section>
`;

exports[`<Login> should render without crashing when login failed 1`] = `
<section
  className="how login-wrapper bg-grey top-space"
  data-pg-name="How it Works"
>
  <div
    className="wrapper"
    data-pg-name="Wrapper"
  >
    <form
      className="card login-form"
      noValidate={true}
      onSubmit={[MockFunction]}
    >
      <header
        className="form-header"
      >
        <h1
          className="text-brand"
        >
          Welcome Back
        </h1>
        <p>
          Sign in to start reporting...
        </p>
      </header>
      <Alert
        message={
          Array [
            "failed",
          ]
        }
        success={false}
        title="Login Failed!"
      />
      <div
        className="form-wrapper"
      >
        <label
          htmlFor="username"
        >
          Username, Email or Phone number
        </label>
        <input
          className="form-element"
          id="username"
          name="username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-wrapper"
      >
        <label
          htmlFor="forgotPassword"
        >
          Password
        </label>
        <Link
          className="forgot-password transition"
          replace={false}
          to="#"
        >
          Forgot Password?
        </Link>
        <input
          autoComplete="off"
          className="form-element"
          id="forgotPassword"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn-brand btn-submit"
        disabled={false}
        type="submit"
      >
        <span>
          Sign In 
        </span>
      </button>
    </form>
    <p
      className="login-register-link"
    >
      New to iReporter?
       
      <Link
        replace={false}
        to="/sign-up"
      >
        Register here
      </Link>
    </p>
  </div>
</section>
`;

exports[`<Login> should render without crashing when login is successful 1`] = `
<section
  className="how login-wrapper bg-grey top-space"
  data-pg-name="How it Works"
>
  <div
    className="wrapper"
    data-pg-name="Wrapper"
  >
    <form
      className="card login-form"
      noValidate={true}
      onSubmit={[MockFunction]}
    >
      <header
        className="form-header"
      >
        <h1
          className="text-brand"
        >
          Welcome Back
        </h1>
        <p>
          Sign in to start reporting...
        </p>
      </header>
      <Alert
        message={
          Array [
            "success",
          ]
        }
        success={true}
        title="Login Successful!"
      />
      <div
        className="form-wrapper"
      >
        <label
          htmlFor="username"
        >
          Username, Email or Phone number
        </label>
        <input
          className="form-element"
          id="username"
          name="username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-wrapper"
      >
        <label
          htmlFor="forgotPassword"
        >
          Password
        </label>
        <Link
          className="forgot-password transition"
          replace={false}
          to="#"
        >
          Forgot Password?
        </Link>
        <input
          autoComplete="off"
          className="form-element"
          id="forgotPassword"
          name="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn-brand btn-submit"
        disabled={true}
        type="submit"
      >
        <span>
          Sign In 
        </span>
        <img
          alt="loading"
          src={Object {}}
          width="12"
        />
      </button>
    </form>
    <p
      className="login-register-link"
    >
      New to iReporter?
       
      <Link
        replace={false}
        to="/sign-up"
      >
        Register here
      </Link>
    </p>
  </div>
</section>
`;
